app/src/lib/stores/app-store.ts

```diff
@@ -2918,11 +2918,10 @@ export class AppStore {
     const branch = tip.branch
     const aheadBehind = state.aheadBehind
 
-    if (!aheadBehind) {
+    if (aheadBehind == null) {
       this._showPopup({
-        type: PopupType.PushBranchCommits,
+        type: PopupType.PublishRepository,
```

Comment:
I'm still a bit unclear on how this fixes the issue. The `aheadBehind` value is associated with the current branch, but we should show the `PublishRepository` popup if there are no remotes associated with the repository. Maybe this fixes the issue because the app will push the current branch after the publish completes?

My gut feeling is that there needs to be two checks here:

 - first, confirm that there's no remote defined -> publish repository flow
 - second, handle situation where branch has no tracking branch -> push the branch to the remote?

Thoughts?

Reply:
> Is it possible to not have a tracking branch at this point?

Good question. I'll see if I can reproduce the issue using the original details and report back.

Reply:
> Is it possible to not have a tracking branch at this point?

An unpublished branch will not have any tracking information. But that should be handled by the old first code path.

Reply:
Shall we move this back to #3583 and get to the bottom of reproducing the issue?